app/controllers/api/comments_controller.rb

```diff
@@ -0,0 +1,63 @@
+class Api::CommentsController < Api::BaseController
+  # Even logged-out users should be able to view comments.
+  skip_before_action :authenticate_user!, only: %i[index]
+
+  def create
+    authorize(Comment)
+
+    comment =
+      Comments::Create.run!(
+        user: current_user,
+        params: comment_params.merge(
+          path: referrer_without_query_params,
+        ),
+      ).comment
+
+    render_schema_json(comment)
+  end
+
+  def destroy
+    comment = current_user.comments.find(params[:id])
+
+    authorize(comment)
+
+    if comment.children.exists?
+      comment.update!(content: '[deleted]', user: nil)
+      render json: comment
```

Comment:
I am not using `render_schema_json` here because I think it would be too difficult/annoying, with the fact that we sometimes respond with comment JSON but sometimes with `head :no_content`.